File: Frontend-Next/src/features/catalogue/domain/entities/sellable-product.ts
/** Units whose quantities are whole numbers (same rule as the backend). */
export const DISCRETE_UNITS: readonly string[] = ['UNIT', 'BOX', 'PACK'];

/** A product as the point of sale sees it: no purchase price, availability at one location. */
export interface SellableProduct {
  readonly id: string;
  readonly sku: string;
  readonly barcode: string | null;
  readonly name: string;
  readonly description: string | null;
  readonly categoryId: string | null;
  readonly categoryName: string | null;
  readonly unit: string;
  readonly salePrice: number;
  readonly batchTracked: boolean;
  readonly availableQuantity: number;
}

export interface Category {
  readonly id: string;
  readonly name: string;
  readonly parentId: string | null;
}

export interface CatalogueQuery {
  readonly locationId: string;
  readonly search?: string;
  readonly categoryId?: string;
  readonly inStock?: boolean;
  readonly page?: number;
  readonly size?: number;
}

File: Backend/src/main/java/com/stockhub/sale/application/dto/SellableProduct.java
package com.stockhub.sale.application.dto;

import java.math.BigDecimal;
import java.util.UUID;

/**
 * A product as the point of sale needs it: sale price and quantity that can be sold at one
 * location. Purchase price and stock settings are deliberately absent.
 *
 * @param availableQuantity stock of the location; for batch-tracked products only non-expired
 *     batches count, as they are the only ones a sale may take
 */
public record SellableProduct(
        UUID id,
        String sku,
        String barcode,
        String name,
        String description,
        UUID categoryId,
        String categoryName,
        String unit,
        BigDecimal salePrice,
        boolean batchTracked,
        boolean hasImage,
        BigDecimal availableQuantity) {}
